### PR TITLE
added journey1a

### PR DESCRIPTION
**Pull request by kashlo**

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -25,6 +25,7 @@
                 android:value="true"/>
 
         <activity
+                android:theme="@style/Theme.AppCompat.Light.NoActionBar"
```
Comment (IgorGanapolsky) [inline app/src/main/AndroidManifest.xml]: Build error: **Cannot resolve symbol '@style/Theme.AppCompat.Light.NoActionBar'**

**Comment by IgorGanapolsky** [comment]
App crashes with:
> Could not find method addTimer(View) in a parent

**Comment by IgorGanapolsky** [comment]
Now that I think of it, Journey 1a should not have ViewPager indicator circles, because user must click **Add Timer** button first:
screen shot 2019-01-17 at 7 41 38 am


What do you think?

File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/AddTimerFragment.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package com.igorganapolsky.vibratingwatchapp
+
+import android.content.Intent
```
Comment (IgorGanapolsky) [inline app/src/main/java/com/igorganapolsky/vibratingwatchapp/AddTimerFragment.kt]: unused import

File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/AddTimerFragment.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+
+class AddTimerFragment : Fragment() {
```
Comment (IgorGanapolsky) [inline app/src/main/java/com/igorganapolsky/vibratingwatchapp/AddTimerFragment.kt]: Let's use Javadocs on public classes as convention

File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/MainActivity.kt [hunk 4]
```diff
@@ -1,15 +1,57 @@
...
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
-        // Enables Always-on
-        setAmbientEnabled()
+        mAmbientController = AmbientModeSupport.attach(this)
+
+        mAdapter =  MyAdapter(supportFragmentManager)
+        val mPager = findViewById<ViewPager>(R.id.pager)
+        mPager.adapter = mAdapter;
```
Comment (IgorGanapolsky) [inline app/src/main/java/com/igorganapolsky/vibratingwatchapp/MainActivity.kt]: no semicolons in Kotlin 😸

File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/MainActivity.kt [hunk 5]
```diff
@@ -1,15 +1,57 @@
...
+import androidx.viewpager.widget.ViewPager
+import androidx.wear.ambient.AmbientModeSupport
+import com.google.android.material.tabs.TabLayout
+import androidx.fragment.app.FragmentPagerAdapter
 
-class MainActivity : WearableActivity() {
+class MainActivity : AppCompatActivity(), AmbientModeSupport.AmbientCallbackProvider {
+
+    private lateinit var mAmbientController: AmbientModeSupport.AmbientController
+    var mAdapter: MyAdapter? = null
```
Comment (IgorGanapolsky) [inline app/src/main/java/com/igorganapolsky/vibratingwatchapp/MainActivity.kt]: Property 'mAdapter' could be private

File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/MainActivity.kt [hunk 6]
```diff
@@ -1,15 +1,57 @@
...
+        }
+
+        override fun onExitAmbient() {
+        }
+
+        override fun onUpdateAmbient() {
+        }
+    }
+
+    class MyAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {
```
Comment (IgorGanapolsky) [inline app/src/main/java/com/igorganapolsky/vibratingwatchapp/MainActivity.kt]: For separation of concerns, would it be better to move this adapter to an external class?

File: app/src/main/res/layout/fragment_add_timer.xml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        android:layout_height="40dp"
+        android:id="@+id/imageButton"
+        android:layout_width="40dp"
+        android:scaleType="centerInside"
+        android:layout_margin="15dp"
+        android:onClick="addTimer"/>
+    <TextView
+      android:layout_width="wrap_content"
+      android:layout_height="wrap_content"
+      android:text="Add timer" android:fontFamily="@font/roboto_condensed_regular" android:textColor="@color/white"/>
```
Comment (IgorGanapolsky) [inline app/src/main/res/layout/fragment_add_timer.xml]: Let's externalize strings

Comment (IgorGanapolsky) [reply app/src/main/res/layout/fragment_add_timer.xml]: This should be two lines

**Comment by kashlo** [comment]
> Now that I think of it, Journey 1a should not have ViewPager indicator circles, because user must click **Add Timer** button first:
> screen shot 2019-01-17 at 7 41 38 am
> 
> What do you think?

We can make the first fragment separate activity, so there would be no dots on the first screen. And only after tapping + button user will be taken to next activity with pager. So user can swipe between fragments if he wants to change settings, but not able to swipe to first screen with + button.

**Comment by IgorGanapolsky** [comment]
Perfect

**Comment by IgorGanapolsky** [comment]
Question: are we only allowing hour/minute timer, or hour/minute/second
screenshot_1547778799

File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/CreateTimerFragmentAdaptor.kt [hunk 8]
```diff
@@ -0,0 +1,23 @@
+package com.igorganapolsky.vibratingwatchapp
+
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentPagerAdapter
+
+class CreateTimerFragmentAdaptor(fm: FragmentManager) : FragmentPagerAdapter(fm) {
```
Comment (IgorGanapolsky) [inline app/src/main/java/com/igorganapolsky/vibratingwatchapp/CreateTimerFragmentAdaptor.kt]: If possible, put Javadoc on public class

File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/AddTimerFragment.kt [hunk 9]
```diff
@@ -0,0 +1,22 @@
+package com.igorganapolsky.vibratingwatchapp
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+
+/**
+ * Returns an AddTimerFragment object
```
Comment (IgorGanapolsky) [inline app/src/main/java/com/igorganapolsky/vibratingwatchapp/AddTimerFragment.kt]: 😉 It doesn't **return**, it **creates a fragment for adding a new timer**

File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/SetBuzzFragment.kt [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+class SetBuzzFragment : Fragment() {
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_set_buzz, container, false)
+    }
+
+
```
Comment (IgorGanapolsky) [inline app/src/main/java/com/igorganapolsky/vibratingwatchapp/SetBuzzFragment.kt]: Unnecessary whitespace

File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/SetTimeFragment.kt [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+import android.view.View
+import android.view.ViewGroup
+import android.widget.NumberPicker
+
+/**
+ * Returns an SetTimeFragment object
+ */
+class SetTimeFragment : Fragment() {
+
+    lateinit var rootView: View
```
Comment (IgorGanapolsky) [inline app/src/main/java/com/igorganapolsky/vibratingwatchapp/SetTimeFragment.kt]: Property 'rootView' could be private

File: app/src/main/java/com/igorganapolsky/vibratingwatchapp/SetTimeFragment.kt [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.NumberPicker
+
+/**
+ * Returns an SetTimeFragment object
```
Comment (IgorGanapolsky) [inline app/src/main/java/com/igorganapolsky/vibratingwatchapp/SetTimeFragment.kt]: .... that does what?  My proposal:
> Creates a Fragment that allows setting the time for a timer.

File: app/src/main/res/layout/fragment_set_buzz.xml [hunk 13]
```diff
@@ -0,0 +1,156 @@
...
+                        android:text="@string/buzz1"
+                        android:textColor="@color/white"
+                        android:textSize="12sp" android:fontFamily="@font/roboto_condensed_regular"/>
+                    <TextView
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:text="@string/buzz1_description"
+                        android:textColor="@color/white"
+                        android:textSize="10sp"
+                        android:layout_marginLeft="3dp" android:fontFamily="@font/roboto_condensed_regular"/>
```
Comment (IgorGanapolsky) [inline app/src/main/res/layout/fragment_set_buzz.xml]: These are two separate properties, belongs on two lines.

File: app/src/main/res/layout/fragment_set_buzz.xml [hunk 14]
```diff
@@ -0,0 +1,156 @@
...
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:text="@string/buzz2"
+                        android:textColor="@color/white" android:fontFamily="@font/roboto_condensed_regular"/>
+                    <TextView
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:text="@string/buzz2_description"
+                        android:textColor="@color/white"
+                        android:textSize="10sp"
```
Comment (IgorGanapolsky) [inline app/src/main/res/layout/fragment_set_buzz.xml]: Is 10sp the recommended size for text on Android Wear? https://designguidelines.withgoogle.com/wearos/style/typography.html

What is your opinion on that?
